core/lib/include/assert.h

```diff
@@ -67,7 +67,7 @@ extern "C" {
 #endif /*__NORETURN*/
 
 #ifdef NDEBUG
-#define assert(ignore)((void)0)
+#define assert(cond) (cond)?((void)0):__builtin_unreachable()
```

Comment:
are all compilers we are targeting supporting the `__builtin_unreachable()` ?

Reply:
CI says yes. I chose it over `__builtin_assume(cond)` which is only supported by Clang